examples/server/app/surface3d/surface3d.ts

```diff
@@ -81,11 +81,12 @@ export class Surface3dView extends LayoutDOMView {
   get_data(): vis.DataSet {
     const data = new vis.DataSet()
     const source = this.model.data_source
+    const data_dict = source.data as Record<string, number[]>
```

Comment:
You can't just assume this. Depending on the context, `data` may be provided as a plain object (`Record<K, V>`) or a mapping (`Map<K, V>`). You have to use appropriate APIs to deal with this. In this situation you can either use `ColumnarDataSource.get()` like this:
```ts
const x = source.get(this.model.x)
//
x[i]
```
or
```ts
import {dict} from "core/util/object"
const data_dict = dict(source.data)
//
data_dict.get(this.model.x)[i]
```